synthesis/synthesis.go

```diff
@@ -342,78 +244,66 @@ func FixCds(sqlitePath string, sequence string, codontable codon.Table, problema
 			if !validBias {
 				return sequence, []Change{}, fmt.Errorf("Invalid bias. Expected NA, GC, or AT, got %s", suggestion.Bias)
 			}
-			// First, let's insert the suggestions that we found using our problematicSequenceFuncs
-			_, err := db.Exec(`INSERT INTO suggestedfix(step, start, end, gcbias, quantityfixes, suggestiontype) VALUES (?, ?, ?, ?, ?, ?)`, i, suggestion.Start, suggestion.End, suggestion.Bias, suggestion.QuantityFixes, suggestion.SuggestionType)
-			if err != nil {
-				return sequence, []Change{}, fmt.Errorf("Failed to insert suggestedfix (does start+end point to valid parts of the amino acid sequence?). Got SQL error: %s", err)
-			}
-		}
 
-		// The following statements are the magic sauce that makes this all worthwhile.
-		// Parameters: step, gcbias, start, end, quantityfix
-		// This query searches for all codons within a certain range (start->end),
-		// organizing all possible changes (from one codon to another) in that range.
-		// If the codon has been used previously, it is removed from the results.
-		// Then, the potential changes are sorted by their potential weight. Weight
-		// is calculated by the frequency of codon use in that organism. Finally, the
-		// top X number of from->to codon pairings are selected and inserted into the
-		// history of the sequence. From now on, the new pairing is used when
-		// selecting the sequence.
-
-		sqlFix1 := `INSERT INTO history -- Top level query inserts fixes into the history
-		            (codon,
-		             pos,
-		             step,
-			     suggestedfix)
-		SELECT codon, pos, step, suggestedfix FROM (SELECT t.codon, -- Secondary level query groups by position and limits the number of changes we do
-		       t.pos,
-		       ? AS step,
-		       ? AS suggestedfix,
-		       t.weight AS weight
-		FROM   (SELECT cb.tocodon AS codon, -- Bottom level query gets all positions that would be interesting given our constraints of position and bias, then organizes them by weight.
-		               s.pos      AS pos,
-			       w.weight AS weight,
-			       h.step AS step
-		        FROM   seq AS s
-		               JOIN history AS h
-		                 ON h.pos = s.pos
-		               JOIN codon AS c
-		                 ON h.codon = c.codon
-		               JOIN codonbias AS cb
-		                 ON cb.fromcodon = c.codon
-			       JOIN 'weights' AS w
-				 ON w.codon = cb.tocodon
-		        WHERE `
-		sqlFix2 := ` s.pos >= ?
-		               AND s.pos <= ?
-		               AND h.codon != cb.tocodon) AS t
-			LEFT JOIN (SELECT codon, pos FROM history WHERE pos >= ? AND pos <= ?) AS his ON t.codon = his.codon AND t.pos = his.pos
-					WHERE his.codon IS NULL
-					ORDER  BY t.step DESC, t.weight DESC) as t
-		GROUP  BY t.pos
-		LIMIT  ?; `
+			// For each suggestion, range between the values
+			var potentialChanges []Change
+			for positionSelector := suggestion.Start; positionSelector <= suggestion.End; positionSelector++ {
```

Comment:
Better commenting for this control flow. Comments should make this easy to read at a glance.